### ISSUE TITLE
[BUG] Matches do not match with number of writes entire

### ISSUE THREAD

**Issue by HuoHua2019** [Chia-Network/chia-blockchain#2249]
**Describe the bug**
The software is plotting, while stopping at an exception that shows: Matches do not match with number of write entries 4295148282 4295148217.

**Expected behavior**
I want to know why would this error happen and how to avoid this.

**Screenshots**
![微信图片_20210421145726](https://user-images.githubusercontent.com/57466413/115511078-087a2380-a2b3-11eb-8da9-cc7802d79687.jpg)

**Desktop (please complete the following information):**

- OS: [Win10]
- Storage: [NVME 2T]

**Additional context**
I am plotting several and only this occurs error.

**Comment by garyhkirk** [comment]
**Describe the bug**

I get this error too (There may be a clue to the cause as the number of write entries fails just after the upper limit of a 32-bit unsigned integer). 

Additionally, the "RuntimeError: Unknown exception" doesn't seem to clear up after itself - the plot still shows as plotting and cannot be deleted in the GUI.

**Expected Behaviour**
Don't know, the first plot I have attempted

**ScreenShots**
<img width="727" alt="ChiaError" src="https://user-images.githubusercontent.com/8881383/115696985-108d9e00-a35b-11eb-8e91-8d8843940d39.png">

**Hardware**
Windows 10 Pro
NVME 1TB & 2TB HDD
80GB RAM
Ryzen 5800X

**Additional context**
Bitfield plotting was enabled and memory was set to 6750

**Comment by evansrightherecoding** [comment]
same error here. I stucked at 11% as well. 
I was using an external nmve drive enclosure. It got fixed after I installed it on the board as an internal drive.
It feels like the error is related to macOS's usb hard drive sleep mechanism.

**Comment by garyhkirk** [comment]
I have had a different problem with the same machine that seems to be a hardware/filesystem issue. I am now plotting on a different drive and will report back with the results

**Comment by hoffmang9** [comment]
These are permissions or storage hardware issues.

**Comment by j4ys0n** [comment]
in what way? why would this just start failing?

**Comment by hoffmang9** [comment]
`Caught plotting error: Matches do not match with number of write entries 4295144356 4295144255` is caused usually by a filesystem but often by bad RAM.

**Comment by henryZhangDev** [comment]
meet the same error, any fix

**Comment by gzaxel** [comment]
the same here :/

**Comment by fiscocsis** [comment]
Same error. It´s my first plotting and it was stucked at 16%... I lost 16h of work...

**Comment by bulusite** [comment]
meet the same error,chia 1.1.5
@hoffmang9 how to fix it

**Comment by Unknown-Kush** [comment]
Same problem.
Caught plotting error: Matches do not match with number of write entries 4295010788 4295010795
Brand new Samsung Evo plus 1TB, 32gb ram, Ryzen 3600.

**Comment by joelwebb** [comment]
I am running into the same issue on Windows (chia 1.1.4) if I plot a single plot k-32 on a hard drive selecting the plot in parallel without any actual parallel plotting. If I run a single plot per SSD using the "Add plot to Queue" instead of "Plot in Parallel" it stopped throwing the errors around 11% on new SSDs.

**Comment by gzaxel** [comment]
I have noticed that problem disappeared when I separated my plotting from farming machine. 
Also I have changed Python version from 3.9.4 to 3.8.5 but no idea whether it had an impact or not.
If you have only one computer you can try virtualization. Anyway it is just more guessing than real solution.

**Comment by room101-dev** [comment]
chia1_1_.log:Caught plotting error: Matches do not match with number of write entries 4292221214 429221325

Linux, Aorus x590 mobo 32gb, 3x evo-970 
Same problem here, here's the known-knowns

This problem only started when I jumped to version 14 ( now on 16 ), the problem started seriously 2-3 days ago, I have tried running machine without the block-chain running ( pure chia plot create ) just a few plots

New machine, New memory, 3 NVME's, all new. Been 'plotting' for a month, so this is a new problem. Sadly there is little to no diagnostics, it's understood about the mismatch, but perhaps the software needs to follow where the divergence first appeared. Always have been staggering, the output above is from 'plotman' format logs.

Running linux mint 19, I do notice that when not running the "npm run electron&" ( blockchain ) there is slightly less failed plots. ( but note that the blockchain is ran on a completely different device, and is low usage of intensive resources )

In summary, this is a NEW problem, didn't exist prior to 2-3 days ago. I find it sad that the 'owner' here just blows this off as a 'memory problem' ( check prior bug reports for months ), sure you can close this off, but how about fixing the problem??

I can't believe for all the people reporting this problem, that it can't be duplicated by DEV???

Have torn down and tested all memory, have tested NVME's; This is a software problem, or a timing problem in the software synchronization ; 

It would be a good test for people that have held back on updates to report whether staying with verisons prior to '13' alleviates this problem, it might be that all new 'chia plot create' past 14 is hosed.

I don't see this as  'memory problem' given that it is new, and widely reported. It's time for DEV to duplicate this problem inhouse.

**Comment by room101-dev** [comment]
> I have noticed that problem disappeared when I separated my plotting from farming machine.
> Also I have changed Python version from 3.9.4 to 3.8.5 but no idea whether it had an impact or not.
> If you have only one computer you can try virtualization. Anyway it is just more guessing than real solution.

I have turned off the blockchain, and I can only say 'slightly less failures' I run on the cmd-line linux, usually 3 plots per NVME, been doing this 1+ month no problem

Just yesterday I ran for 24 hours on this machine with 'npm run electron' not ran, and I still got errors 'mismatch'

**Comment by crisperhills** [comment]
> `Caught plotting error: Matches do not match with number of write entries 4295144356 4295144255` is caused usually by a filesystem but often by bad RAM.

This started occurring for me after I installed new RAM modules, lending more support to the notion that this is related to malfunctioning or bad RAM. I haven't yet had the opportunity to run a full torture test on the new modules to confirm, but the behavior I've observed falls right in line -- no issue before slotting in new RAM, rampant issues after.

I've since pulled the new modules and restarted plotting to determine if things proceed smoothly from now on. I'll update here with the results thereof. Nevertheless, I'd recommend anyone running into this error run a full check of their RAM (with, say, Memtest86/+).

**Comment by mirushaki** [comment]
The same error for me when trying to plot in parallel.
CPU: AMD Ryzen 5 2600
RAM: 32GB (4x8gb)
NVME: HP EX950 2tb
HARD DRIVES: 2x 14tb external USB hard drives
OS: kubuntu 21.04

Plotting in sequence doesn't cause any error, but plotting in parallel (even 2 plots in parallel) randomly generates this error. And after this error, the entire system becomes unresponsive.
I checked RAM modules with memtest86+, no errors.

**Comment by crisperhills** [comment]
> I've since pulled the new modules and restarted plotting to determine if things proceed smoothly from now on. I'll update here with the results thereof.

Plotting has been proceeding without issue after removing the new RAM modules.

**Comment by link04200** [comment]
I’m having this issue aswell. I recently upgraded to 64gb of ram and I could plot 3 in parallel for days with no issues. I then added a new nvme m.2 to see how many more I could plot in parallel and since then I’ve been getting errors like crazy even when I go back to plotting 3 on the original drive.

**Comment by 42x42x42** [comment]
changed RAM, now all seems to be fine

**Comment by prophetz2021** [comment]
same problem here. I already done 33 plots with my 4 sticks of 4GB ram(16 GB) on windows 10 without error at all .but when i reinstall windows the problem is occure. I stuck at 6% in plotting. with this error message 

Total matches: 4292724138
Caught plotting error: Matches do not match with number of write entries 4292724138 4292719018
[1188] Failed to execute script chia
Traceback (most recent call last):
  File "chia\cmds\chia.py", line 81, in <module>
  File "chia\cmds\chia.py", line 77, in main
  File "click\core.py", line 829, in __call__
  File "click\core.py", line 782, in main
  File "click\core.py", line 1259, in invoke
  File "click\core.py", line 1259, in invoke
  File "click\core.py", line 1066, in invoke
  File "click\core.py", line 610, in invoke
  File "click\decorators.py", line 21, in new_func
  File "chia\cmds\plots.py", line 135, in create_cmd
  File "chia\plotting\create_plots.py", line 176, in create_plots
RuntimeError: Unknown exception

i tried many way to fix it but can't solve the problem. so sad

**Comment by prophetz2021** [comment]
Hey guys i think i have a solution for this error.i can plot 100% complete now. all i did is format a drive that use for temp. 
"Full format not quick format". i also change a sata cable too. i use a same ram stick as before, everything work fine. 
just try it guys hope this can help you all.

**Comment by Timpetu81** [comment]
Issue exists in 1.1.6 on Windows. Machine is Z440 (E5-1650v4) with 4 x 8 GB ECC and Samsung 1,9 TB PM1643a on Dell H200 HBA.

As all parts are enterprise grade I do not expect a memory or storage problem. There were no issue till i tried parallel plotting the first time now. The delay to the next plotting start was choosen too short, so the number of plots increased too much over time. Memory usage was max 16 GB and storage left 120 GB at least. Not sure if this storage "shortage" could cause the problem - it was the ~10th plot that throw the error.

To prevent those overplotting issues, instead of a time delay we should implement a percentage delay to the progress of the last active plot (or the lowest progress of all active plottings) - default value could be 20 - 35 %. I now it is individual anyway, but in my opinion better reference to run-time instead of real-time.

Caught plotting error: Matches do not match with number of write entries 4112248740 4112248742
[3140] Failed to execute script chia
Traceback (most recent call last):
  File "chia\cmds\chia.py", line 81, in <module>
  File "chia\cmds\chia.py", line 77, in main
  File "click\core.py", line 829, in __call__
  File "click\core.py", line 782, in main
  File "click\core.py", line 1259, in invoke
  File "click\core.py", line 1259, in invoke
  File "click\core.py", line 1066, in invoke
  File "click\core.py", line 610, in invoke
  File "click\decorators.py", line 21, in new_func
  File "chia\cmds\plots.py", line 135, in create_cmd
  File "chia\plotting\create_plots.py", line 176, in create_plots
RuntimeError: Unknown exception

**Comment by bigdog001** [comment]
@prophetz2021  have you fixed this since you done the full disk re-format ? i am facing the same issue now , sadly experience!

**Comment by gowthamv93** [comment]
I get this error when plotting on SAS 15k 300GB drives on Ubuntu OS but not getting error with the same machine / same drives running Windows OS. Must be a problem with recent chia version?

**Comment by madmmmax** [comment]
Still having this issue.
5900x
x570e
64g 2666
1tb nvme*3

**Comment by tkmbe99** [comment]
Yesterday same problem again with 1.1.7

**Comment by ishtob** [comment]
still occurring on ubuntu with 1.2.0

here's my error log:

`Total matches: 8570247200
Caught plotting error: Matches do not match with number of write entries 8570247200 8570215498
Traceback (most recent call last):
  File "/home/ishtob/chia-blockchain/venv/bin/chia", line 33, in <module>
    sys.exit(load_entry_point('chia-blockchain', 'console_scripts', 'chia')())
  File "/home/ishtob/chia-blockchain/chia/cmds/chia.py", line 79, in main
    cli()  # pylint: disable=no-value-for-parameter
  File "/home/ishtob/chia-blockchain/venv/lib/python3.9/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/ishtob/chia-blockchain/venv/lib/python3.9/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/ishtob/chia-blockchain/venv/lib/python3.9/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/ishtob/chia-blockchain/venv/lib/python3.9/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/ishtob/chia-blockchain/venv/lib/python3.9/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/ishtob/chia-blockchain/venv/lib/python3.9/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/ishtob/chia-blockchain/venv/lib/python3.9/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/ishtob/chia-blockchain/chia/cmds/plots.py", line 135, in create_cmd
    create_plots(Params(), ctx.obj["root_path"])
  File "/home/ishtob/chia-blockchain/chia/plotting/create_plots.py", line 168, in create_plots
    plotter.create_plot_disk(
RuntimeError: std::exception`

**Comment by hoffmang9** [comment]
These are almost always hardware or system configuration issues. It's why this is a closed issue.
